### DISCUSSION TITLE
'reconstructPar' does not work if only Adjoint is run

### DISCUSSION THREAD

**Discussion by Mohsin-Bashir** [mdolab/dafoam#556] [General]
I ran the runScript_v2.py with runAdjoint and "writeSensMap": ["shapey"], in daOptions, for NACA0012_Airfoil-incompressible case.
The error looks like as follows: 
```
dafoamuser@53471128f62b:~/mount$ reconstructPar
/*---------------------------------------------------------------------------*\
| =========                 |                                                 |
| \\      /  F ield         | OpenFOAM: The Open Source CFD Toolbox           |
|  \\    /   O peration     | Version:  v1812                                 |
|   \\  /    A nd           | Web:      www.OpenFOAM.com                      |
|    \\/     M anipulation  |                                                 |
\*---------------------------------------------------------------------------*/
Build  : v1812 OPENFOAM=1812
Arch   : "LSB;label=32;scalar=64"
Exec   : reconstructPar
Date   : Jan 08 2024
Time   : 01:12:32
Host   : 53471128f62b
PID    : 643
I/O    : uncollated
Case   : /home/dafoamuser/mount
nProcs : 1
trapFpe: Floating point exception trapping enabled (FOAM_SIGFPE).
fileModificationChecking : Monitoring run-time modified files using timeStampMaster (fileModificationSkew 10)
allowSystemOperations : Allowing user-supplied system call operations

// * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * //
Create time

Overriding DebugSwitches according to controlDict
    SolverPerformance 0;

Overriding DebugSwitches according to controlDict
    SolverPerformance 0;

Overriding DebugSwitches according to controlDict
    SolverPerformance 0;

Overriding DebugSwitches according to controlDict
    SolverPerformance 0;

Overriding DebugSwitches according to controlDict
    SolverPerformance 0;



Reconstructing fields for mesh region0

Time = 0.00000000001



--> FOAM FATAL IO ERROR:
unexpected class name dictionary expected faceCompactList or faceList
    while reading object faces

file: /home/dafoamuser/mount/processor0/0.00000000001/polyMesh/faces at line 15.

    From function void Foam::CompactIOList<T, BaseType>::readFromStream() [with T = Foam::face; BaseType = int]
    in file lnInclude/CompactIOList.C at line 48.

FOAM exiting
```

**Comment by friedenhe** [comment]
Thanks for the feedback. The writeSensMap is not tested and may be broken. We will check what causes this.

**Reply by Mohsin-Bashir** [reply]
Would you be able to provide an estimated timeline for when the 'writeSensMap' issue is expected to be resolved? I appreciate your attention to this matter.

**Comment by friedenhe** [comment]
We just fixed the issue for the sensMap. Please check this commit: aa18fb4231b6ee5169e56fc27bcd44f1e33afb06

**Reply by Mohsin-Bashir** [reply]
"`reconstructPar`" is working perfectly now.

But there is an issue. Previously, the code used to generate "`boundary, faces, neighbour, owner, points`" in this folder `0.00000000001\polyMesh` and "`sens_CD_shapey`" in `0.00000000001`.
`sens_CD_shapey` are sensitivities at face centers of design surface.
But I want to calculate sensitivities at each node of design surface. So, to calculate node(point) sensitivities, I wrote a python script, I used these files faces, points and sens_CD_shapey and generated a file like this one 
[IDCoordSens.txt](https://github.com/mdolab/dafoam/files/14071406/IDCoordSens.txt). In this file, first column is for node ID (for all nodes on design surface), then coordinates of that node and finally node sensitivity.

Since the code no longer generates these files, could you please guide me how can I get this information (IDCoordSens.txt), I mean which files/functions need to be changed.

**Comment by friedenhe** [comment]
Now the sensitivity is written in a format fully consistent with other field variables such as U and p. You can get the faces, neighbour, and owner information from the `mesh` object. This information is well documented on CFD-online.com. You can also google it.

**Reply by Mohsin-Bashir** [reply]
Got it. Thank you so much.
